linkerd/app/inbound/src/lib.rs

```diff
@@ -279,13 +272,16 @@ impl Config {
         svc::stack(http_profile_cache)
             .push_on_response(svc::layers().box_http_response())
             .push_make_ready()
-            // Don't resolve profiles for inbound-targetted requests. They'll be
-            // resolved on the outbound side if necsesary.
-            .push(admit::AdmitLayer::new(prevent_loop))
             .push_fallback(
                 http_target_cache
                     .push_on_response(svc::layers().box_http_response().box_http_request()),
             )
+            .push(admit::AdmitLayer::new(prevent_loop))
+            .push_fallback_on_error::<prevent_loop::LoopPrevented, _>(
```

Comment:
May be worth a comment explaining that this has to be outside the cache? Not a blocker.